Harmony/Public/Harmony.cs

```diff
@@ -210,52 +210,62 @@ public static MethodInfo ReversePatch(MethodBase original, HarmonyMethod standin
 			return PatchFunctions.ReversePatch(standin, original, transpiler, null);
 		}
 
+		/// <summary>Unpatches all methods that were patched by the specified <paramref name="harmonyID"/>. Unpatching is done by repatching methods without patches of this instance.</summary>
+		/// <param name="harmonyID">The Harmony ID to restrict unpatching to a specific Harmony instance.</param>
+		public static void UnpatchID(string harmonyID)
+		{
+			if (string.IsNullOrEmpty(harmonyID))
```

Comment:
Since this is a new method altogether, does it not make sense to throw an `ArgumentNullException` in this case? `null` isn't a valid argument value for `harmonyID` now.

Reply:
I don't mind changing it to throw an `ArgumentNullException`, but I just return and log [as per @ghorsington's suggestion in PR #40 ](https://github.com/BepInEx/HarmonyX/pull/40#issuecomment-997419874). Granted, that was regarding the `UnpatchAll` method.

Reply:
+1 for the exception, but the case for an empty string should also be included. Anything that's an invalid ID should not be allowed to continue

Reply:
Changed to throw an `ArgumentNullException` on both `null` and empty `harmonyID`'s.